### DISCUSSION TITLE
Forwarding RealTime Messages

### DISCUSSION THREAD

**Discussion by ndsizeif** [FortySevenEffects/arduino_midi_library#217] [Q&A] [answered]
Greetings, 
I have been using code based on the DualMerger.ino example in the repository in a couple of projects.  It works very well.  In my last project I had a device that cared about real time clock, start, and stop messages.  The code I was using from that sketch doesn't seem to forward them.  

void loop()
      
      
        {
      
      
            if (midiA.read())
      
      
            {
      
      
                // Thru on A has already pushed the input message to out A.
      
      
                // Forward the message to out B as well.
      
      
                midiB.send(midiA.getType(),
      
      
                           midiA.getData1(),
      
      
                           midiA.getData2(),
      
      
                           midiA.getChannel());
      
      
            }
I'm simply trying to get the real time messages that are coming in from midiA, to get sent to midiB just like note on/off, program change, and control change are.  After reading the documentation I'm confident that I am making the solution more complicated than it is in reality.  Real time messages get transmitted just fine with midi thru.  I suppose the merger code is just missing a function.  I tried to use getTransport(), but that isn't it.

If someone could point me the right direction I would appreciate it.  Thank you.

**Comment by lathoub** [comment, marked as answer]
Hi

The above code will not work, if you want to send RealTime messages (typically 1 byte long). The `midiB.send` command will filter out any Real Time messages (only channel messages are processed).

https://github.com/FortySevenEffects/arduino_midi_library/blob/d9149d19df867abbf78cfcd37ef9d0e14dedca7f/src/MIDI.hpp#L159-L166

You might want to use `midiB.sendRealTime` command (and check first if the incoming data from `midiA` is a RealTime message)

**Reply by ndsizeif** [reply]
I appreciate you allowing me to figure out a solution while giving a bit of help.
I needed to read the MIDI.hpp file you linked more closely.


    if (MIDIA.getType() >= 0xF0) {
    
        MIDIB.sendRealTime(MIDIA.getType());

This got me the results I wanted.  Thanks again.
